File: leetcode-swift/lc_027.swift
//
// LeetCode
// 27. Remove Element
// https://leetcode.com/problems/remove-element/
//


class Solution {
    func removeElement(_ nums: inout [Int], _ val: Int) -> Int {
        var currentIndex = 0
        var lastIndex = nums.count - 1

        while currentIndex <= lastIndex {
            if nums[currentIndex] == val {
                nums[currentIndex] = nums[lastIndex]
                lastIndex -= 1
            } else {
                currentIndex += 1
            }
        }

        return lastIndex + 1
    }
}

File: leetcode-go/lc_027.go
//
// LeetCode
// 27. Remove Element
// https://leetcode.com/problems/remove-element/
//


func removeElement(nums []int, val int) int {
    currentIndex := 0
    lastIndex := len(nums) - 1

    for currentIndex <= lastIndex {
        if nums[currentIndex] == val {
            nums[currentIndex] = nums[lastIndex]
            lastIndex--
        } else {
            currentIndex++
        }
    }

    return lastIndex + 1
}
